contracts/strategies/NFTVoteWinnerTakeAll.sol

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.19;
+
+import { ERC721 } from "@solady/tokens/ERC721.sol";
+import { ERC20 } from "@solady/tokens/ERC20.sol";
+import { BaseStrategy } from "./BaseStrategy.sol";
+import { Transfer } from "../core/libraries/Transfer.sol";
+
+contract NFTVoteWinnerTakeAll is BaseStrategy {
+    uint startTime;
+    uint endTime;
+    ERC721 nft;
+    address winner;
+
+    address[] memory recipients;
+    mapping(address => bool) public isRecipient;
+    mapping(uint => bool) public isVoted;
+    mapping(address => uint) public votes;
+
+    function initialize(bytes32 _identityId, uint256 _poolId, bytes memory _data) external override {
+        super.initialize(_identityId, _poolId, _data);
+        (nft, startTime, endTime) = ERC721(abi.decode(_data, (address, uint, uint)));
```

Comment:
i've never seen this syntax with parsing the whole abi.decode. does it actually works? does it also work if two addresses are included?